src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java

```diff
@@ -76,6 +76,10 @@
      */
     private static final ThreadLocal<List<Entry>> ENTRIES = new ThreadLocal<>();
 
+    static {
+        ENTRIES.set(new LinkedList<Entry>());
```

Comment:
I think you should set the initial value by subclassing ThreadLocal above; there is an example in the Javadoc:

https://docs.oracle.com/javase/7/docs/api/java/lang/ThreadLocal.html

If you set the ThreadLocal value in the static block I _believe_ that it will set the value only for the thread loading the class.